cls/SourceControl/Git/Settings.cls

```diff
@@ -141,6 +183,51 @@ Method %Save() As %Status
     quit $$$OK
 }
 
+Method ToDynamicObject() As %DynamicObject
+{
+    // uses custom methods rather than %JSON.Adaptor because Mappings multidimensional
+    // array is not supported
+    set settingsJSON = {
+        "pullEventClass": (..pullEventClass),
+        "percentClassReplace": (..percentClassReplace),
+        "Mappings": {}
+    }
+    do settingsJSON.%Set("decomposeProductions",..decomposeProductions,"boolean")
+    set k1 = $order(..Mappings(""))
+    while (k1 '= "") {
+        do settingsJSON.Mappings.%Set(k1, {})
+        set k2 = $order(..Mappings(k1,""))
+        while (k2 '= "") {
+            do settingsJSON.Mappings.%Get(k1).%Set(k2,{"directory": (..Mappings(k1,k2))})
+            if $get(..Mappings(k1,k2,"NoFolders")) {
+                do settingsJSON.Mappings.%Get(k1).%Get(k2).%Set("noFolders", 1, "boolean")
+            }
+            set k2 = $order(..Mappings(k1, k2))
+        }
+        set k1 = $order(..Mappings(k1))
+    }
+    return settingsJSON
+}
+
+Method ImportDynamicObject(pSettingsDyn As %DynamicObject)
+{
+    set ..pullEventClass = pSettingsDyn.%Get("pullEventClass")
+    set ..percentClassReplace = pSettingsDyn.%Get("percentClassReplace")
+    set ..decomposeProductions = pSettingsDyn.%Get("decomposeProductions")
+    kill ..Mappings
+    set mappingsDyn = pSettingsDyn.%Get("Mappings", {})
+    set i1 = mappingsDyn.%GetIterator()
+    while i1.%GetNext(.k1, .v1) {
+        set i2 = v1.%GetIterator()
+        while i2.%GetNext(.k2, .v2) {
+            set ..Mappings(k1, k2) = v2.%Get("directory")
+            if v2.%Get("noFolders") {
+                set ..Mappings(k1, k2, "NoFolders") = 1
+            }
```

Comment:
This needs an else to clear the flag if NoFolders changes - otherwise we could get stuck at 1